contracts/contract/src/main.rs

```diff
@@ -2727,35 +2569,37 @@ fn install_contract() {
 
     let named_keys = {
         let mut named_keys = NamedKeys::new();
-        named_keys.insert(INSTALLER.to_string(), runtime::get_caller().into());
+        named_keys.insert(INSTALLER.to_string(), Key::Account(runtime::get_caller()));
 
         named_keys
     };
 
     let hash_key_name = format!("{PREFIX_HASH_KEY_NAME}_{collection_name}");
+    let message_topics = BTreeMap::from([(EVENTS.to_string(), MessageTopicOperation::Add)]);
 
     let (contract_hash, contract_version) = storage::new_contract(
         entry_points,
         Some(named_keys),
         Some(hash_key_name.clone()),
         Some(format!("{PREFIX_ACCESS_KEY_NAME}_{collection_name}")),
+        Some(message_topics),
     );
 
-    // Store contract_hash and contract_version under the keys CONTRACT_NAME and CONTRACT_VERSION
+    // Store contract_hash and contract_version under account keys
     runtime::put_key(
         &format!("{PREFIX_CONTRACT_NAME}_{collection_name}"),
-        contract_hash.into(),
+        Key::Hash(contract_hash.value()),
     );
+
     runtime::put_key(
         &format!("{PREFIX_CONTRACT_VERSION}_{collection_name}"),
-        storage::new_uref(contract_version).into(),
+        storage::new_uref(get_contract_version_key(contract_version).to_string()).into(),
     );
 
-    let nft_contract_package_hash: ContractPackageHash = runtime::get_key(&hash_key_name)
-        .unwrap_or_revert()
-        .into_hash()
-        .map(ContractPackageHash::new)
-        .unwrap();
+    let nft_contract_package_hash: PackageHash = runtime::get_key(&hash_key_name)
```

Comment:
Can we write a contract package hash key instead of a package key?
(like in CEP18)